bld/namelist_files/namelist_definition_ctsm.xml

```diff
@@ -1549,6 +1554,60 @@ Mapping method from Nitrogen deposition input file to the model resolution
     copy     = copy using the same indices
 </entry>
 
+<!-- ========================================================================================  -->
+<!-- FAN Namelist (only used when bgc=cn/cndv and use_fan=.true.)                 -->
+<!-- ========================================================================================  -->
+
+<entry id="stream_year_first_fan" type="integer" category="datasets"
+       group="fan_nml" valid_values="" >
+First year to loop over for FAN Nitrogen (manure) Deposition data
+</entry>
+
+<entry id="stream_year_last_fan" type="integer" category="datasets"
+       group="fan_nml" valid_values="" >
+Last year to loop over for FAN Nitrogen (manure) Deposition data
+</entry>
+
+<entry id="model_year_align_fan" type="integer" category="datasets"
+       group="fan_nml" valid_values="" >
+Simulation year that aligns with stream_year_first_ndep2 value
+</entry>
+
+<entry id="stream_fldfilename_fan" type="char*256" category="datasets"
+       input_pathname="abs" group="fan_nml" valid_values="" >
+Filename of input stream data for FAN Nitrogen (manure) Deposition
+</entry>
+
+<entry id="fan_mapalgo" type="char*256" category="datasets"
+       group="fan_nml" valid_values="bilinear,nn,nnoni,nnonj,spval,copy" >
+Mapping method from FAN Nitrogen (manure) deposition input file to the model resolution
```

Comment:
So the model will interpolate from the resolution on the file to the model resolution using bilinear interpolation. Here you could change it to something like nearest neighbor or a straight copy if it's at the same resolution. If you have some data at different resolutions, you would need to have more than one file for it. So you could add a PH streams file for example that was at a finer resolution say.

Putting this on streams files allows you to control the horizontal mapping as well as time-interpolation. So it's a general mechanism that doesn't require custom code, but gives the user a lot of control. So I think it's the best method for moving forward...

Reply:
I have the data for up to 0.25 degree resolution. I don't terribly much like the linear interpolation, because it doesn't conserve the total. Also, because some of the maps are very patchy and you can have locally big errors if you interpolate, say, from 0.25 to 2 degrees. Maybe it is not possible to support every possible grid, but I'd try to avoid linear interpolation between very different resolutions.

There are also some issues with missing values in ocean as seen for pH...